packages/pluggableWidgets/datagrid-web/src/components/ColumnSelector.tsx

```diff
@@ -19,11 +20,40 @@ export function ColumnSelector(props: ColumnSelectorProps): ReactElement {
 
     useOnClickOutside([buttonRef, optionsRef], () => setShow(false));
 
+    const onClick = useCallback(
+        (isVisible: boolean, id: string) =>
+            props.setHiddenColumns(prev => {
+                if (!isVisible) {
+                    prev.splice(
+                        prev.findIndex(v => v === id),
+                        1
+                    );
+                    return [...prev];
+                } else {
+                    return [...prev, id];
+                }
+            }),
+        [props.setHiddenColumns]
+    );
+
+    const firstHidableColumnIndex = useMemo(() => props.columns.findIndex(c => c.canHide), [props.columns]);
+    const lastHidableColumnIndex = useMemo(() => {
+        const index = props.columns
+            .slice()
+            .reverse()
+            .findIndex(c => c.canHide);
```

Comment:
I feel like 
1. applying slice to go over the entire array and copy, 
2. applying reverse to go over the array and reverse its order, 
3. go over the arrya again to find the appropriate index, 
4. calculate the length of the array to inverse the index

Isn't quite a great solution performance wise and also readability wise. Maybe one of the following would be nicer:

1. Create a util function that takes an array and a condition function, and uses a for loop to go over the array back to front and returns the element if the condition is true or `-1` if not. Basically our own (efficient) implementation of a `findLastIndex`.
2. `const index = props.column.map(c => c.canHide).lastIndexOf(true);`, Then update the rest of this callback accordingly.

Reply:
I liked number 2, gonna apply it.